scijava/scijava-ops-engine/src/test/java/org/scijava/ops/engine/OpMethodDependencyPositionTest.java

```diff
@@ -17,82 +21,93 @@
 public class OpMethodDependencyPositionTest extends AbstractTestEnvironment
 		implements OpCollection {
 
-	@BeforeAll
-	public static void addNeededOps() {
-		ops.register(new OpMethodDependencyPositionTest());
-	}
-
-	@OpField(names = "test.stringToLong")
-	public final Function<String, Long> parser = in -> Long.parseLong(in);
-
-	@OpMethod(names = "test.dependencyBeforeOutput",
-		type = Computers.Arity1.class)
-	public static void OpMethodFoo(List<String> in, @OpDependency(
-		name = "test.stringToLong") Function<String, Long> op, List<Long> out)
-	{
+	public static void goodDep( //
+		@OpDependency(name = "someDep") Function<String, Long> op, //
+		List<String> in, //
+		List<Long> out //
+	) {
 		out.clear();
 		for (String s : in)
 			out.add(op.apply(s));
 	}
 
 	@Test
-	public void testOpDependencyBeforeOutput() {
-		List<String> in = new ArrayList<>();
-		in.add("1");
-		List<Long> out = new ArrayList<>();
-		ops.op("test.dependencyBeforeOutput").arity1().input(in).output(out).compute();
-		List<Long> expected = Arrays.asList(1l);
-		assertIterationsEqual(expected, out);
+	public void testOpDependencyBefore() throws NoSuchMethodException {
+		var m = this.getClass().getDeclaredMethod(//
+			"goodDep", //
+			Function.class, //
+			List.class, //
+			List.class //
+		);
+		var info = new OpMethodInfo( //
```

Comment:
I don't really like seeing `OpMethodInfo` manually constructed outside of a test to explicitly make sure `OpMethodInfo` construction works as intended. My intuition is that ops tests should go through an `OpCollection` as they would be used in production.

Reply:
> I don't really like seeing `OpMethodInfo` manually constructed outside of a test to explicitly make sure `OpMethodInfo` construction works as intended.

Well, it is *in* a test :sweat_smile: 

> My intuition is that ops tests should go through an `OpCollection` as they would be used in production.

If we were testing an actual Op that would be used in production, I'd agree, but the thing we're trying to test here (i.e. ensuring that the `OpMethodInfo` is invalid *because* of the position of the `OpDependency`) would be impossible to test if we went through the `OpBuilder` in some way. This is also the minimal amount of code needed to test this behavior - i.e. it's a unit test.

For the above reasons, I believe that it should be left the way it is. I suppose we could refactor this into a test that ensures improper Ops are not matched, but I think that way of writing tests is much more prone to error.

With that said, we generally *don't* test like this, and there is something to be said for consistency. If you really think consistency is important, feel free to rewrite the test like I described, or in your own preferred way!

Reply:
@gselzer It is fine to have a focused unit test for `OpInfo` creation but it is not sufficient. There also need to be framework-level tests that show how invalid ops are handled.

I added my thoughts after trying to tweak these tests to https://github.com/scijava/scijava/issues/25. So, it's fine to wait until that issue is addressed to make more changes.